packages/datadog-plugin-connect/src/index.js

```diff
@@ -38,7 +40,7 @@ function createWrapUse () {
       const layer = this.stack[index]
 
       if (layer && layer.handle) {
-        this.stack[index].handle = wrapLayerHandle(layer)
+        this.stack[index].handle = wrapLayerHandle(layer, config)
```

Comment:
Would it require less passing around to provide it to `web.patch` instead and store it on the request? It would also make the config available to other web helpers if needed.

Reply:
I tend to like the idea of explicitly passing the config down to where it's required rather than relying on some other bit of instrumentation to make it available? It also make it a bit easier for me to follow logically. If you feel like this is a blocker I'm open to revamping it, it just may take me a bit of time to refactor everything.

Reply:
One of the main goals of the web util is to remove as much tracing code as possible from the integrations and instead attach everything to the request. Considering that, I think it would be better to set the config as soon as possible so that it doesn't need to be passed again everywhere that it's needed.

Reply:
alright, fair enough, will try to knock out this change

Reply:
well, it would appear i am obtuse, as the config already exists on `req._datadog` by the time the request is passed into the wrapMiddleware function.

with `web.instrument`, immediately after .patch is called [here](https://github.com/DataDog/dd-trace-js/blob/564f0782327620243a78ee5a544248a37458acd9/packages/dd-trace/src/plugins/util/web.js#L52-L54) we invoke `startSpan` which sets `_datadog.config` [here](https://github.com/DataDog/dd-trace-js/blob/564f0782327620243a78ee5a544248a37458acd9/packages/dd-trace/src/plugins/util/web.js#L211)

So all this passing down of `config` appears to have been totally unnecessary on my part. That being said, I figure it can't hurt to at least explicitly define an empty config object in the  `patch` call on the request's `_datadog` property, which would make any later calls to check `req._datadog.config` a bit more defensive since worst case the config value would be undefined. What do you think? I think this is much cleaner now for sure

Reply:
i suppose i could also just explicitly pass in config to `.patch` and set it within `patch`, if that just simplifies things even more

Reply:
Do you mean for `getMiddlewareSetting`? If that's the case, I think for now whatever is simplest. Eventually we should probably do this in a safe and consistent way.

Reply:
no not particularly, `getMiddlewareSetting` is for ensuring the config.middleware is normalized and if not set, defaults to `true`. Similar to `getStatusValidator`

What i was referring to is that I was passing config down all the way into every invocation of `web.wrapMiddleware` like [here](https://github.com/DataDog/dd-trace-js/blob/5348eaad6f1a86d57b87ffd3de6397feb1dcb417/packages/datadog-plugin-router/src/index.js#L116) , but since `web.instrument` already handles setting the `_datadog.config` object (before `web.wrapMiddleware` would get invoked), we can just reference `req._datadog.config.middleware` within `web.wrapMiddleware` without having to go through all the hassle of passing down config into it explicitly. Does that make sense?

Reply:
Yes, and I'd say to definitely reuse the existing config.